Sources/XCTest/Public/XCTestMain.swift

```diff
@@ -58,17 +58,58 @@
 ///     ./FooTests FooTestCase          # Run all the tests in FooTestCase
 ///
 /// - Parameter testCases: An array of test cases run, each produced by a call to the `testCase` function
-/// - seealso: `testCase`
+/// - seealso: `testCase
+
+//public func XCTMain<T: XCTestCase>(
+//    _ testCases: [XCTRunnableTestCase]
+//) -> Never {
+//    let arguments = CommandLine.arguments
+//    let executionMode = ArgumentParser(arguments: arguments).executionMode
+//
+//    let rootTestSuite: XCTestSuite
+//    let currentTestSuite: XCTestSuite
+//    if executionMode.selectedTestNames == nil {
+//        rootTestSuite = XCTestSuite(name: "All tests")
+//        currentTestSuite = XCTestSuite(name: "\(Bundle.main.bundleURL.lastPathComponent).xctest")
+//        rootTestSuite.addTest(currentTestSuite)
+//    } else {
+//        rootTestSuite = XCTestSuite(name: "Selected tests")
+//        currentTestSuite = rootTestSuite
+//    }
+//
+//    testCases.map(XCTestCaseSuite.init).forEach(currentTestSuite.addTest)
+//
+//    handleExecutionMode(executionMode,
+//                        rootTestSuite: rootTestSuite,
+//                        arguments: arguments,
+//                        observers: [])
+//}
+
+//@available(*, deprecated, message: "Use XCTMain(_ runnableTests:) instead.")
```

Comment:
So, I went and added them throughout the stack as it cleaned up a lot of spurious project warnings where we create a deprecation warning in our project because we're using the deprecated API. 👼